### ISSUE TITLE
TensorRT runs much slower for non-continuous inference

### ISSUE THREAD

**Issue by Apeiria** [NVIDIA/TensorRT#2107]
## Description

When inserting sleep(); between inference, the inference time becomes much longer
``` c++
    for (int i = 0; i < 10000; i++)
    {
        auto t1 = high_resolution_clock::now();
        context->executeV2(bindings);
        // context->enqueueV2(bindings, stream, nullptr);
        // cudaStreamSynchronize(stream);
        auto t2 = high_resolution_clock::now();
        cout<<"iter " << i << " time = " << duration_cast<microseconds>(t2 - t1).count() / 1000.0 << " ms" << endl;
        _sleep(1000);
    }
```
![image](https://user-images.githubusercontent.com/60050293/176581386-c2ebe458-876e-40e7-aa4d-cdf3b3dd2268.png)

If the `_sleep(1000);` is removed, the inference time will be very stable, roughly 2ms, no matter how long it runs.

## Environment

**TensorRT Version**: 8.4.1.5
**NVIDIA GPU**: RTX 2070 super
**NVIDIA Driver Version**: 516.01
**CUDA Version**: 11.7
**CUDNN Version**: 8.4.1.50
**Operating System**: windows 11
**Python Version (if applicable)**: 
**Tensorflow Version (if applicable)**: 
**PyTorch Version (if applicable)**: 
**Baremetal or Container (if so, version)**: 

## Steps To Reproduce
Load any tensorrt model, compare the following for loop with/without sleep();
```
for(...)
{
    t1 = clock();
    inference();
    t2 = clock();
    print(t2-t1);
    sleep();
}
```

**Comment by nvpohanh** [comment]
This is usually caused by GPUs going to idle and not an issue in TRT. Could you check the solution in https://github.com/NVIDIA/TensorRT/issues/2042 and see if it works?

**Comment by Apeiria** [comment]
> This is usually caused by GPUs going to idle and not an issue in TRT. Could you check the solution in #2042 and see if it works?

Thanks for your reply, it works! However are there anything I can do in the c++ code to prevent the GPU from going to sleep when idle?

**Comment by nvpohanh** [comment]
No, it's mostly caused by the GPU driver, so you need to change the driver settings.

**Comment by Apeiria** [comment]
> No, it's mostly caused by the GPU driver, so you need to change the driver settings.

OK, thanks a lot for your help!

**Comment by Apeiria** [comment]
> No, it's mostly caused by the GPU driver, so you need to change the driver settings.

Hi again,
I just did the same experiment in pytorch, and I found the inference time is quite stable regardless of how long I make it sleep. Do you have any idea why the behavior is different for pytorch and tensorrt?

``` python
    for j in range(10000):
        with torch.inference_mode():
            t1 = time.time()
            pred = model(img)
            print("iter", j, "time =", (time.time()-t1)*1000,"ms")
        time.sleep(1)
```
![image](https://user-images.githubusercontent.com/60050293/176602170-8219caef-a25a-4fb1-a01b-5904d6c59409.png)


Thanks!

**Comment by nvpohanh** [comment]
@Apeiria Is the PyTorch using GPU or CPU? Also, could you try `time.sleep(5)` and see if it makes any difference?

**Comment by Apeiria** [comment]
> @Apeiria Is the PyTorch using GPU or CPU? Also, could you try `time.sleep(5)` and see if it makes any difference?

My pytorch is using gpu.
I tried time.sleep(5) and it seems the same, below is the result.
![image](https://user-images.githubusercontent.com/60050293/176665796-5bd838f2-e1d3-4a19-93a8-c10595c01d45.png)

**Comment by nvpohanh** [comment]
interesting... we will take a look when having the chance. Meanwhile, I still recommend that you try the approaches in https://github.com/NVIDIA/TensorRT/issues/2042 to see if any of them fixes the issue.

**Comment by haibozhang123** [comment]
I fix this question, but it cost lot of time  

so i need tensorrt time give me reward ，to announce the solution

mailbox ： 780437510@qq.com

**Comment by ttyio** [comment]
closing issues no activity for more than 3 weeks per our policy, thanks all!
